src/st/no0/5BDCC.c

```diff
@@ -1,8 +1,134 @@
+        entity->flags = entity->flags | FLAG_UNK_2000;
+        DestroyEntity(self);
+        return;
+    }
+
+    switch (self->step) {
+    case OUIJA_TABLE_INIT:
+        InitializeEntity(D_us_80180C14);
+        self->animCurFrame = 5;
+        self->hitboxOffY = 2;
+        return;
+    case OUIJA_TABLE_INIT_SUBENTITIES:
+        if (UnkCollisionFunc3(D_us_801825D4) & 1) {
+            // Spawn objects on the table
+            entity = VASE;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, entity);
+            entity->params = 0;
+            entity->ext.ouijaTableContents.parent = self;
+            entity->posX.i.hi -= 4;
+            entity->posY.i.hi -= 16;
+
+            entity = BEAKER;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, entity);
+            entity->params = 1;
+            entity->ext.ouijaTableContents.parent = self;
+            entity->posX.i.hi += 2;
+            entity->posY.i.hi -= 20;
+
+            entity = WALKING_TABLE;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, entity);
+            entity->params = 2;
+            entity->ext.ouijaTableContents.parent = self;
+
+            if (self->params) {
+                entity->posX.i.hi += 24;
+                entity->facingLeft = 1;
+            } else {
+                entity->posX.i.hi -= 24;
+            }
+#ifndef VERSION_PSP
+            // Looks like a coding error
+            entity->posY.i.hi = entity->posY.i.hi;
+#endif
+            self->step++;
+        }
+        break;
+    case OUIJA_TABLE_COME_ALIVE:
+        // When player approaches, spring to life
+        if ((GetDistanceToPlayerX() < 80) && (GetDistanceToPlayerY() < 64)) {
+            self->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+            entity = BEAKER;
+            entity->ext.ouijaTableContents.spawned = true;
+
+            entity = VASE;
+            entity->ext.ouijaTableContents.spawned = true;
+            SetStep(3);
+        }
+        break;
+    case OUIJA_TABLE_ATTACK:
+        if (!self->step_s) {
+            self->ext.ouijaTable.attackTimer = 128;
+            self->step_s += 1;
+        }
+        if (AnimateEntity(D_us_801825F4, self) == 0) {
+            self->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+        }
+        UnkCollisionFunc2(D_us_801825CC);
+
+        if (self->facingLeft) {
+            self->velocityX = FIX(0.5);
+        } else {
+            self->velocityX = FIX(-0.5);
+        }
+
+        if (!(--self->ext.ouijaTable.attackTimer)) {
+            self->ext.ouijaTable.attackTimer = 192;
+            // Throw out one of the objects on the table
+            if (self->ext.ouijaTable.thrownVase != 0) {
+                entity = VASE;
+            } else {
+                entity = BEAKER;
+            }
+            entity->ext.ouijaTableContents.isThrown = true;
+            // Alternate between throwing vase and beaker
+            self->ext.ouijaTable.thrownVase ^= 1;
```

Comment:
Wasn't super confident with the naming here, 
I think it's basically a toggle between which sub-entity gets thrown but wasn't sure what to call it

Reply:
maybe leave it as an unk for now; circling back later it's a lot easier to identify something that's unfinished (and finish it) than to discover that something is wrong.